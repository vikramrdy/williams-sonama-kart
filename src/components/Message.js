import { makeStyles, Snackbar } from "@material-ui/core";
import MuiAlert from "@material-ui/lab/Alert";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";

function Alert(props) {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    "& > * + *": {
      marginTop: theme.spacing(2),
    },
  },
}));

const Message = ({ message, showMessage, severity }) => {
  const classes = useStyles();
  const [open, setOpen] = useState(showMessage);

  useEffect(() => {
    setOpen(showMessage);
  }, [showMessage]);

  const handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }

    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <Snackbar open={open} autoHideDuration={6000} onClose={handleClose}>
        <Alert onClose={handleClose} severity={severity}>
          {message}
        </Alert>
      </Snackbar>
    </div>
  );
};

Message.propTypes = {
  message: PropTypes.string.isRequired,
  severity: PropTypes.oneOf(["error", "warning", "success", "info"]),
  showMessage: PropTypes.bool,
};

Message.defaultProps = {
  severity: "info",
  showMessage: false,
};

export default Message;
