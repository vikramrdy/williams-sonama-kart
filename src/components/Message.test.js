import { render } from "@testing-library/react";
import React from "react";

import Message from "./Message";

describe("Message Component", () => {
  it("should match the snapshot", () => {
    const modalProps = {
      showMessage: true,
      message: "Image not present",
    };
    const { asFragment } = render(<Message {...modalProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the info message", () => {
    const modalProps = {
      showMessage: true,
      message: "Image not present",
    };
    const { getByText } = render(<Message {...modalProps} />);
    expect(getByText(/Image not present/i)).toBeInTheDocument();
  });

  it("should not render the message component", () => {
    const modalProps = {
      showMessage: false,
      message: "Image not present",
    };
    const { queryByText } = render(<Message {...modalProps} />);
    expect(queryByText(/Image not present/i)).toBeFalsy();
  });
});
