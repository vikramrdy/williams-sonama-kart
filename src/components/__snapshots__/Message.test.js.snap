// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation6"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Image not present
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            title="Close"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
